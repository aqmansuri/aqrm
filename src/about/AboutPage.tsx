import {
  IonButton,
  IonButtons,
  IonContent,
  IonHeader,
  IonIcon,
  IonItem,
  IonLabel,
  IonList,
  IonNote,
  IonPage,
  IonTitle,
  IonToolbar,
} from '@ionic/react';
import { logOutOutline } from 'ionicons/icons';
import { useHistory } from 'react-router';
import { name, author, description, version } from '../../package.json';
import { useAuthentication } from '../core/auth';

const AboutPage: React.FC = () => {
  const { logout } = useAuthentication();
  const history = useHistory();

  const handleLogout = async () => {
    await logout();
    history.replace('/login');
  };

  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonTitle>About Tea Taster</IonTitle>
          <IonButtons slot="end">
            <IonButton
              data-testid="logout-button"
              onClick={() => handleLogout()}
            >
              <IonIcon slot="icon-only" icon={logOutOutline} />
            </IonButton>
          </IonButtons>
        </IonToolbar>
      </IonHeader>
      <IonContent className="main-content">
        <IonHeader collapse="condense">
          <IonToolbar>
            <IonTitle size="large">About Tea Taster</IonTitle>
          </IonToolbar>
        </IonHeader>
        <IonList>
          <IonItem>
            <IonLabel>Name</IonLabel>
            <IonNote slot="end">{name}</IonNote>
          </IonItem>
          <IonItem>
            <IonLabel>Description</IonLabel>
            <IonNote slot="end">{description}</IonNote>
          </IonItem>
          <IonItem>
            <IonLabel>Version</IonLabel>
            <IonNote slot="end">{version}</IonNote>
          </IonItem>
          <IonItem>
            <IonLabel>Author</IonLabel>
            <IonNote slot="end">{author.name}</IonNote>
          </IonItem>
        </IonList>
      </IonContent>
    </IonPage>
  );
};
export default AboutPage;
