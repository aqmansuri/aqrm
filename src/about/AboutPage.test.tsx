import { render } from '@testing-library/react';
import AboutPage from './AboutPage';

describe('<AboutPage />', () => {
  it('renders consistently', () => {
    const { asFragment } = render(<AboutPage />);
    expect(asFragment).toMatchSnapshot();
  });
});
