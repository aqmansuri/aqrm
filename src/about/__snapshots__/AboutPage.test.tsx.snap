// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage /> renders consistently 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-title>
          About Tea Taster
        </ion-title>
        <ion-buttons
          slot="end"
        >
          <ion-button
            data-testid="logout-button"
          >
            <ion-icon
              icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Log Out</title><path d='M304 336v40a40 40 0 01-40 40H104a40 40 0 01-40-40V136a40 40 0 0140-40h152c22.09 0 48 17.91 48 40v40M368 336l80-80-80-80M176 256h256' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
              slot="icon-only"
            />
          </ion-button>
        </ion-buttons>
      </ion-toolbar>
    </ion-header>
    <ion-content
      class="main-content"
    >
      <ion-header
        collapse="condense"
      >
        <ion-toolbar>
          <ion-title
            size="large"
          >
            About Tea Taster
          </ion-title>
        </ion-toolbar>
      </ion-header>
      <ion-list>
        <ion-item>
          <ion-label>
            Name
          </ion-label>
          <ion-note
            slot="end"
          >
            tea-taster
          </ion-note>
        </ion-item>
        <ion-item>
          <ion-label>
            Description
          </ion-label>
          <ion-note
            slot="end"
          >
            Tea Tasting Notes
          </ion-note>
        </ion-item>
        <ion-item>
          <ion-label>
            Version
          </ion-label>
          <ion-note
            slot="end"
          >
            1.0.0
          </ion-note>
        </ion-item>
        <ion-item>
          <ion-label>
            Author
          </ion-label>
          <ion-note
            slot="end"
          >
            Ionic
          </ion-note>
        </ion-item>
      </ion-list>
    </ion-content>
  </div>
</DocumentFragment>
`;
